### DISCUSSION TITLE
is there a restorefaces option

### DISCUSSION THREAD

**Discussion by anonmgit** [comfyanonymous/ComfyUI#158] [Q&A]
Automatic1111 has a `restorefaces` option as well. I've managed to replicate all my setup (and it runs 2x faster on comfy). Is there an equivalent of the `restorefaces` option from automatic1111 that we can use here? 
I was unable to find anything in issues or discussions

**Comment by WASasquatch** [comment]
There isn't. And I honestly wonder if they would even be appropriate to use in software like this. Last I heard their datasets were research only and can't be used for anything commercial because of CelebsA dataset and GFPGANs Chinese citizens.

**Reply by anonmgit** [reply]
I was under assumption that comfyui isn't comercial software. Its option would be useful and based on user's usecase. If someone is using this comercially then it can be disabled (maybe disable by default)

**Comment by comfyanonymous** [comment]
I think the UpscaleModelLoader can load the face restore models but I don't think ImageUpscaleWithModel will handle them correctly so I would need to implement another node for that.

**Reply by WASasquatch** [reply]
> I think the UpscaleModelLoader can load the face restore models but I don't think ImageUpscaleWithModel will handle them correctly so I would need to implement another node for that.

GFPGAN at least can easily be implemented with a def function, CodeFormer is more walled off and gotta send data to it's repo folder structure, and fetch it out. Hopefully they are done messing with it to, cause on Easy Diffusion they kept breaking it downstream with changes constantly and it became an annoyance. Like nitpicking over the name of variables being used (bad juju in publicly updated code)

**Reply by comfyanonymous** [reply]
The code I took for the models: https://github.com/comfyanonymous/ComfyUI/tree/master/comfy_extras/chainner_models/architecture/face

Is from chainner and I think they did a good job implementing the models in a clean way.

I'm not familiar with the details though because face restoration is not something I use.

**Reply by WASasquatch** [reply]
> The code I took for the models: https://github.com/comfyanonymous/ComfyUI/tree/master/comfy_extras/chainner_models/architecture/face
> 
> Is from chainner and I think they did a good job implementing the models in a clean way.
> 
> I'm not familiar with the details though because face restoration is not something I use.

Oh awesome! They stripped out codeformer into a single file like I've been doing for some nodes that are based on repos. Looks good.

I also don't use face enhance, hr-fix is enough for me. I don't like the generic look they give. Also the license issues with the datasets usage is an issue for my work.
